src/dataworker/PoolRebalanceUtils.ts

```diff
@@ -233,16 +233,12 @@ export async function subtractExcessFromPreviousSlowFillsFromRunningBalances(
   await Promise.all(
     allValidFillsInRange
       .filter((fill) => {
-        // @todo This filter implicitly produces an array of v2 fills because it is impossible for v3 fills to pass.
-        // Update the filter such that it also passes v3 fills where a slow fill was initially produced.
         const outputAmount = sdkUtils.getFillOutputAmount(fill);
         const fillAmount = sdkUtils.getFillAmount(fill);
         const totalFilledAmount = sdkUtils.getTotalFilledAmount(fill);
         return totalFilledAmount.eq(outputAmount) && !fillAmount.eq(outputAmount);
       })
-      .map(async (fill: interfaces.FillWithBlock) => {
-        assert(sdkUtils.isV2Fill(fill)); // @todo Remove when the above filter permits v3 fills to pass.
-
+      .map(async (fill: interfaces.V2FillWithBlock) => {
```

Comment:
Can we get away without the type annotation?

```suggestion
      .map(async (fill) => {
```